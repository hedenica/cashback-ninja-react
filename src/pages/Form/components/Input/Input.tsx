import { InputHTMLAttributes } from 'react'
import clsx from 'clsx'
import styles from './input.module.scss'

interface InputProps extends InputHTMLAttributes<HTMLInputElement> {
  className?: string;
  type: string;
}

export function Input ({
  className,
  type,
  ...remainingProps
}: InputProps) {
  return (
    <input
      className={clsx(className, styles.input)}
      type={type}
      {...remainingProps}
    />
  )
}
