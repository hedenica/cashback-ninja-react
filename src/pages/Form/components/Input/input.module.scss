.input {
  border: 2px solid var(--yellow);
  border-radius: 4px;
  height: 3rem;
  width: 250px;
  padding: 10px;
  background: inherit;
  caret-color: var(--yellow);
}

.input::placeholder {
  color: var(--yellow);
  letter-spacing: 0.122rem;
}