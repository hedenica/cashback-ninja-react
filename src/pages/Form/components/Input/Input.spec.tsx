import { render } from '@testing-library/react'
import { Input } from './Input'

describe('Input', () => {
  it('should render properly', () => {
    const { container } = render(
      <Input
        type='text'
        placeholder='foo'
      />)

    expect(container).toMatchSnapshot()
  })
})
