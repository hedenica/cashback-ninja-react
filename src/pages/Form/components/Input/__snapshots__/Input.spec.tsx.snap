// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render properly 1`] = `
<div>
  <input
    class="input"
    placeholder="foo"
    type="text"
  />
</div>
`;
